### ISSUE TITLE
grpclb_end2end_test failure

### ISSUE THREAD

**Issue by markdroth** [grpc/grpc#11089]
https://grpc-testing.appspot.com/job/gRPC_master_tsan_cpp/1457/testReport/junit/(root)/c++_linux_tsan_native/bins_tsan_grpclb_end2end_test___gtest_filter_SingleBalancerTest_Vanilla__GRPC_POLL_STRATEGY_epoll/

Output:

```
I0511 14:37:59.222790804    9845 ev_epoll_linux.c:95]        epoll engine will be using signal: 40
D0511 14:37:59.223388421    9845 ev_posix.c:107]             Using polling engine: epoll
D0511 14:37:59.224404397    9845 dns_resolver.c:316]         Using native dns resolver
D0511 14:37:59.224500122    9845 test_config.c:403]          test slowdown factor: sanitizer=5, fixture=1, poller=1, total=5
Note: Google Test filter = SingleBalancerTest.Vanilla
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from SingleBalancerTest
[ RUN      ] SingleBalancerTest.Vanilla
I0511 14:37:59.233376428    9845 grpclb_end2end_test.cc:410] starting backend server on port 26859
I0511 14:37:59.235422569    9926 server_builder.cc:258]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0511 14:37:59.277781591    9845 grpclb_end2end_test.cc:417] backend server startup complete
I0511 14:37:59.280528179    9845 grpclb_end2end_test.cc:410] starting backend server on port 2116
I0511 14:37:59.281756693    9969 server_builder.cc:258]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0511 14:37:59.318836859    9845 grpclb_end2end_test.cc:417] backend server startup complete
I0511 14:37:59.322123232    9845 grpclb_end2end_test.cc:410] starting backend server on port 29440
I0511 14:37:59.323389219    9987 server_builder.cc:258]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0511 14:37:59.359263968    9845 grpclb_end2end_test.cc:417] backend server startup complete
I0511 14:37:59.362089020    9845 grpclb_end2end_test.cc:410] starting backend server on port 11626
I0511 14:37:59.363214964   10009 server_builder.cc:258]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0511 14:37:59.403775868    9845 grpclb_end2end_test.cc:417] backend server startup complete
I0511 14:37:59.407117428    9845 grpclb_end2end_test.cc:410] starting balancer server on port 10984
I0511 14:37:59.408368901   10040 server_builder.cc:258]      Synchronous server. Num CQs: 1, Min pollers: 1, Max Pollers: 2, CQ timeout (msec): 10000
I0511 14:37:59.441150687    9845 grpclb_end2end_test.cc:417] balancer server startup complete
I0511 14:37:59.460542764   10059 grpclb_end2end_test.cc:187] LB: recv msg 'initial_request {
  name: "servername_not_used"
}
'
I0511 14:37:59.460619895   10059 grpclb_end2end_test.cc:265] LB: sleeping for 0 ms...
I0511 14:37:59.460904429   10059 grpclb_end2end_test.cc:268] LB: Woke up! Sending response 'server_list {
  servers {
    ip_address: "\177\000\000\001"
    port: 26859
  }
  servers {
    ip_address: "\177\000\000\001"
    port: 2116
  }
  servers {
    ip_address: "\177\000\000\001"
    port: 29440
  }
  servers {
    ip_address: "\177\000\000\001"
    port: 11626
  }
}
'
I0511 14:37:59.462197219    9845 grpclb.c:511]               Missing LB token for backend address 'ipv4:127.0.0.1:26859'. The empty token will be used instead
I0511 14:37:59.462243678    9845 grpclb.c:511]               Missing LB token for backend address 'ipv4:127.0.0.1:2116'. The empty token will be used instead
I0511 14:37:59.462272807    9845 grpclb.c:511]               Missing LB token for backend address 'ipv4:127.0.0.1:29440'. The empty token will be used instead
I0511 14:37:59.462298463    9845 grpclb.c:511]               Missing LB token for backend address 'ipv4:127.0.0.1:11626'. The empty token will be used instead
test/cpp/end2end/grpclb_end2end_test.cc:478: Failure
Value of: status_and_response.first.ok()
  Actual: false
Expected: true
test/cpp/end2end/grpclb_end2end_test.cc:484: Failure
      Expected: 100
To be equal to: backend_servers_[i].service_->request_count()
      Which is: 124
test/cpp/end2end/grpclb_end2end_test.cc:484: Failure
      Expected: 100
To be equal to: backend_servers_[i].service_->request_count()
      Which is: 124
test/cpp/end2end/grpclb_end2end_test.cc:484: Failure
      Expected: 100
To be equal to: backend_servers_[i].service_->request_count()
      Which is: 0
test/cpp/end2end/grpclb_end2end_test.cc:484: Failure
      Expected: 100
To be equal to: backend_servers_[i].service_->request_count()
      Which is: 123
I0511 14:38:00.408971440    9845 grpclb_end2end_test.cc:434] backend about to shutdown
I0511 14:38:00.410799055    9845 grpclb_end2end_test.cc:437] backend shutdown completed
I0511 14:38:00.410831830    9845 grpclb_end2end_test.cc:434] backend about to shutdown
I0511 14:38:00.415370313    9845 grpclb_end2end_test.cc:437] backend shutdown completed
I0511 14:38:00.415418703    9845 grpclb_end2end_test.cc:434] backend about to shutdown
I0511 14:38:00.416912034    9845 grpclb_end2end_test.cc:437] backend shutdown completed
I0511 14:38:00.416944794    9845 grpclb_end2end_test.cc:434] backend about to shutdown
I0511 14:38:00.418563513    9845 grpclb_end2end_test.cc:437] backend shutdown completed
I0511 14:38:00.418603867    9845 grpclb_end2end_test.cc:434] balancer about to shutdown
I0511 14:38:00.420256558    9845 grpclb_end2end_test.cc:437] balancer shutdown completed
[  FAILED  ] SingleBalancerTest.Vanilla (1210 ms)
[----------] 1 test from SingleBalancerTest (1210 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (1210 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] SingleBalancerTest.Vanilla

 1 FAILED TEST
D0511 14:38:00.435271341    9845 ev_posix.c:107]             Using polling engine: epoll
D0511 14:38:00.435316574    9845 dns_resolver.c:316]         Using native dns resolver
D0511 14:38:00.438284413    9845 ev_posix.c:107]             Using polling engine: epoll
D0511 14:38:00.438349398    9845 dns_resolver.c:316]         Using native dns resolver
D0511 14:38:00.440938902    9845 ev_posix.c:107]             Using polling engine: epoll
D0511 14:38:00.441000112    9845 dns_resolver.c:316]         Using native dns resolver
D0511 14:38:00.447013569    9845 ev_posix.c:107]             Using polling engine: epoll
D0511 14:38:00.447087439    9845 dns_resolver.c:316]         Using native dns resolver
D0511 14:38:00.451094932    9845 ev_posix.c:107]             Using polling engine: epoll
D0511 14:38:00.451169512    9845 dns_resolver.c:316]         Using native dns resolver
Command exited with non-zero status 1
real 1.28
user 0.94
sys 0.21
```

**Comment by nathanielmanistaatgoogle** [comment]
Seen in [gRPC_master_linux #2521](https://grpc-testing.appspot.com/job/gRPC_master_linux/2521/).

**Comment by dgquintas** [comment]
this is fixed in #11159, to be merged soon

**Comment by markdroth** [comment]
David, I think this is fixed now, isn't it?  If so, please close.
